gnmpagerduty/tasks.py

```diff
@@ -0,0 +1,181 @@
+
+    headers['Authorization']="Basic %s" % auth
+    headers['Content-Type']=content_type
+
+    if not re.match(r'^/',urlpath):
+        urlpath = '/' + urlpath
+
+    url = "{0}:{1}{2}".format(settings.VIDISPINE_URL,settings.VIDISPINE_PORT,urlpath)
+    logger.debug("URL is %s" % url)
+    (rtn_headers,content) = agent.request(url,method=method,body=body,headers=headers)
+    if int(rtn_headers['status']) < 200 or int(rtn_headers['status']) > 299:
+        raise HttpError(int(rtn_headers['status']),url,headers,rtn_headers,content)
+    return (rtn_headers,content)
+
+
+def get_system_type():
+    from django.core.cache import cache
+    import httplib2
+    system_type = cache.get('gnmpagerduty_system_type')
+    if system_type is None:
+        h=httplib2.Http()
+        headers, content = make_vidispine_request(h,'GET','/API/site?type=current',"",{'Accept': 'application/json'})
+
+        output = str(content)
+
+        if "KP" in output:
+            system_type = 'Live'
+        elif "BK" in output:
+            system_type = 'DC2'
+        elif "VX" in output:
+            system_type = 'Development'
+        else:
+            system_type = 'Unknown'
+        cache.set('gnmpagerduty_system_type', system_type)
+    return system_type
+
+
+@periodic_task(run_every=(crontab(minute='*/10')), name="check_storage", ignore_result=True)
+def check_storage():
+    from gnmvidispine.vs_storage import VSStoragePathMap
+    from django.conf import settings
+    from models import StorageData, IncidentKeys
+    import requests
+    import json
+
+    logger.info("check_storage run by Celery.")
+
+    i = app.control.inspect()
+
+    running = str(i.active())
+
+    rnumber = running.count("check_storage")
+
+    if rnumber > 1:
+        return "Task aborted to avoid duplication"
+
+    system_type = get_system_type()
+
+    storage_data = VSStoragePathMap(url=settings.VIDISPINE_URL,port=settings.VIDISPINE_PORT,
+                                    user=settings.VIDISPINE_USERNAME,passwd=settings.VIDISPINE_PASSWORD,
+                                    run_as=settings.VIDISPINE_USERNAME)
+    sd = {'data':'value', 'test':'value'}
+    sd['map'] = map(lambda (k,v): v.__dict__, storage_data.items())
+
+    returnthis = "Under threshold:"
+
+    url = 'https://events.pagerduty.com/generic/2010-04-15/create_event.json'
+
+    for val in sd['map']:
+        record = StorageData.objects.get(storage_id=val['name'])
+        try:
+            key = IncidentKeys.objects.get(storage_id=val['name'])
+        except Exception as e:
+            key = IncidentKeys.objects.create(storage_id=val['name'])
+
+        intts = int(record.trigger_size)
+        intfc = int(val['freeCapacity'])
+        hffc = human_friendly(intfc)
+        intc = int(val['capacity'])
+        percent_free_float = (100.0 / intc) * intfc
+        percent_free = int(percent_free_float)
+
```

Comment:
this whole section is quite difficult to read.  Please could you split it into a couple of seperate functions, and use more descriptive variable names than intts.